crates/holochain/src/core/workflow/sys_validation_workflow/unit_tests.rs

```diff
@@ -244,23 +335,38 @@ impl TestCase {
             Arc::new(self.dna_def.clone()),
         );
 
-        let mut network = MockHolochainP2pDnaT::new();
-        // This can't be copied so if you want to call the run twice you would need to reset the network behaviour!
         let actual_network = self
             .actual_network
             .take()
             .unwrap_or_else(|| MockHolochainP2pDnaT::new());
-        network.expect_clone().return_once(move || actual_network);
+
+        let op_sender = IncomingDhtOpSender::new(
+            Arc::new(self.test_space.space.clone()),
+            self.self_trigger.0.clone(),
+        );
 
         sys_validation_workflow(
             Arc::new(workspace),
-            Arc::new(self.test_space.space.clone()),
+            op_sender,
+            self.current_validation_dependencies.clone(),
             self.app_validation_trigger.0.clone(),
             self.self_trigger.0.clone(),
-            network,
+            actual_network,
         )
         .await
+        .unwrap()
+    }
+
+    async fn check_trigger_and_rerun(&mut self) -> WorkComplete {
```

Comment:
For tests that hit the network, they will have to simulate the re-trigger that the workflow will normally do for itself.